### PR TITLE
Update automation to use Go 1.19

### PR DESCRIPTION
**Pull request by lostluck**
Upgrade which version of go we're using in the non-playground automation scripts. Current latest version is 1.19.3.

This doesn't affect the minimum Go version for SDK users, which remains at 1.18

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by lostluck** [comment]
Run Java PreCommit

**Comment by lostluck** [comment]
Run SQL PreCommit

File: .github/workflows/go_tests.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ jobs:
           fetch-depth: 2
       - uses: actions/setup-go@v3
         with:
-          go-version: '1.18'
+          go-version: '1.19'
```
Comment (robertwb) [inline .github/workflows/go_tests.yml:47]: Should we still test on the oldest version of Go that we support?

**Review by robertwb — APPROVED** [review]

**Comment by lostluck** [comment]
Run SQL PreCommit

**Comment by lostluck** [comment]
Run Java PreCommit

**Comment by Abacn** [comment]
Looks like this breaks GitHub Action Go Test due to go fmt: https://github.com/apache/beam/actions/workflows/go_tests.yml

Maybe Go1.19 introduced some different fmt criteria

**Comment by lostluck** [comment]
That does sound familiar TBH. Ok. We should revert the action change only (as that's what changes the fmt requirement we've instituted for beam.

This does mean that we would need to split the fmt check and the unit tests in the action, since of the versions have different fmt requirements, they can't both be satisfied.

File: .github/workflows/go_tests.yml (hunk 1 above)
Comment (lostluck) [reply .github/workflows/go_tests.yml:47]: I replied and wasn't aware of Danny's response. Certainly can't do matrix testing along with the format check as evidenced by the failure with this 1.19 change, but if that's split out, then I don't have a problem with.

I am against doing so at the same time as a routine upgrade. We should file a p3 issue to do the matrix testing change if we're keen on it, and separately for fixing the formatting for 1.19.  1.19 introduces newer interpretations of the Go Doc comments which allows for richer linkified documentation. I'll file the formating fix issue.

**Pull request merged**